isis/src/qisis/apps/ipce/IpceMainWindow.cpp

```diff
@@ -636,49 +672,76 @@ namespace Isis {
    *   @history Ian Humphrey - Settings are now read on a project name basis. References #4358.
    *   @history Tyler Wilson 2017-11-02 - Settings now read recent projects.  References #4492.
    *   @history Tyler Wilson 2017-11-13 - Commented out a resize call near the end because it
-   *                                      was messing with the positions of widgets after a
-   *                                      project was loaded.  Fixes #5075.
+   *                was messing with the positions of widgets after a project was loaded.  
+   *                Fixes #5075.
+   *   @history Makayla Shepherd 2018-06-10 - Settings are read from the project root ipce.config.
+   *                If that does not exist then we read from .Isis/ipce/ipce.config.
    */
   void IpceMainWindow::readSettings(Project *project) {
     // Ensure that the Project pointer is not NULL
     if (!project) {
       QString msg = "Cannot read settings with a NULL Project pointer.";
       throw IException(IException::Programmer, msg, _FILEINFO_);
     }
-
+    
+    // Set the path of the settings file
+    // The default is to assume that the project has an ipce.config in it
+    // If the file does not exist then we read settings from .Isis/ipce/ipce.config
     QString appName = QApplication::applicationName();
-
-    QSettings settings(
-        FileName("$HOME/.Isis/" + appName + "/" + appName + "_" + project->name() + ".config")
-          .expanded(), QSettings::NativeFormat);
-
-    // General settings
+    QString filePath = project->projectRoot() + "/ipce.config";
+    bool isFullScreen = false;
+    if (!FileName(filePath).fileExists()) {
+      filePath = "$HOME/.Isis/" + appName + "/ipce.config";
+      // If the $HOME/.Isis/ipce/ipce.config does not exist then we want ipce to show up in 
+      // in full screen. In other words the default geometry is full screen
+      if (!FileName(filePath).fileExists()) {
+        isFullScreen = true;
+      }
+    }
+    
     if (project->name() == "Project") {
       setWindowTitle("ipce");
+    }
+    else {
+      setWindowTitle( project->name() );
+      QString projName = project->name();
+      setWindowTitle(projName );
+    }
```

Comment:
Does `setWindowTitle( project->name() ) ;` not work directly? If not, can line 706 be removed? It looks to me like lines 707 and 708 do the same thing as 706?

Reply:
:+1: